### ISSUE TITLE
Can not reproduce ODA result

### ISSUE THREAD

**Issue by lwpyh** [tim-learn/SHOT#31]
Hi, here I am reproducing the ODA task on Office-Home dataset, but I cannot achieve the results in the paper, I follow the command in the readme.md, and my result in closed-set are good. I even get less results on source retrain result(e.g on Cl->Rr 42.3(my) vs 49.2(paper)), but I have follow your environment requirements in the readme.md and still not work well. I run it on V110 and 2080Ti but both cannot achieve satisfied results, could you provide the environment list so I can check which part I made a mistake?

Best

**Comment by tim-learn** [comment]
> 

Sorry for the late reply. I use the same code on several different machines, the results varies but the changes are small. The lastest environment is shown in the following,

> # Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             4.5                       1_gnu    defaults
absl-py                   0.14.0                   pypi_0    pypi
addict                    2.4.0                    pypi_0    pypi
apex                      0.1                      pypi_0    pypi
attrs                     21.4.0                   pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
beautifulsoup4            4.10.0                   pypi_0    pypi
blas                      1.0                         mkl    defaults
blessings                 1.7                      pypi_0    pypi
boto3                     1.20.31                  pypi_0    pypi
botocore                  1.23.31                  pypi_0    pypi
bravado                   11.0.3                   pypi_0    pypi
bravado-core              5.17.0                   pypi_0    pypi
ca-certificates           2021.5.30            ha878542_0    conda-forge
certifi                   2021.5.30        py36h5fab9bb_0    conda-forge
charset-normalizer        2.0.7                    pypi_0    pypi
click                     8.0.3                    pypi_0    pypi
colorama                  0.4.4                    pypi_0    pypi
configparser              5.0.2                    pypi_0    pypi
contextlib2               21.6.0                   pypi_0    pypi
cudatoolkit               11.0.3               h15472ef_8    conda-forge
cvxopt                    1.2.7                    pypi_0    pypi
cycler                    0.10.0                   pypi_0    pypi
dalib                     0.2                      pypi_0    pypi
dataclasses               0.8                pyh787bdff_2    conda-forge
decorator                 4.4.2                    pypi_0    pypi
docker-pycreds            0.4.0                    pypi_0    pypi
easydl                    2.0.7                    pypi_0    pypi
efficientnet-pytorch      0.7.1                    pypi_0    pypi
filelock                  3.4.0                    pypi_0    pypi
freetype                  2.10.4               h0708190_1    conda-forge
future                    0.18.2                   pypi_0    pypi
gdown                     4.2.0                    pypi_0    pypi
gitdb                     4.0.9                    pypi_0    pypi
gitpython                 3.1.18                   pypi_0    pypi
gorilla                   0.4.0                    pypi_0    pypi
gorilla-core              0.2.7.8                  pypi_0    pypi
gorilla2d                 0.2.8.7                  pypi_0    pypi
gpustat                   0.6.0                    pypi_0    pypi
idna                      3.3                      pypi_0    pypi
imageio                   2.9.0                    pypi_0    pypi
importlib-metadata        4.8.1                    pypi_0    pypi
iniconfig                 1.1.1                    pypi_0    pypi
intel-openmp              2021.3.0          h06a4308_3350    defaults
ipdb                      0.13.9                   pypi_0    pypi
ipython                   7.16.3                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
jedi                      0.17.2                   pypi_0    pypi
jinja2                    3.0.3                    pypi_0    pypi
jmespath                  0.10.0                   pypi_0    pypi
joblib                    1.0.1                    pypi_0    pypi
jpeg                      9b                   h024ee3a_2    defaults
jsonpointer               2.2                      pypi_0    pypi
jsonref                   0.2                      pypi_0    pypi
jsonschema                3.2.0                    pypi_0    pypi
kiwisolver                1.3.1                    pypi_0    pypi
lcms2                     2.12                 h3be6417_0    defaults
libblas                   3.9.0            11_linux64_mkl    conda-forge
libcblas                  3.9.0            11_linux64_mkl    conda-forge
libedit                   3.1.20210714         h7f8727e_0    defaults
libffi                    3.2.1             hf484d3e_1007    defaults
libgcc-ng                 9.3.0               h5101ec6_17    defaults
libgomp                   9.3.0               h5101ec6_17    defaults
liblapack                 3.9.0            11_linux64_mkl    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libstdcxx-ng              9.3.0               hd4cf53a_17    defaults
libtiff                   4.1.0                h2733197_1    defaults
libuv                     1.42.0               h7f98852_0    conda-forge
littleutils               0.2.2                    pypi_0    pypi
lmdb                      1.3.0                    pypi_0    pypi
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
markupsafe                2.0.1                    pypi_0    pypi
matplotlib                3.3.4                    pypi_0    pypi
mkl                       2021.3.0           h06a4308_520    defaults
ml-collections            0.1.0                    pypi_0    pypi
monotonic                 1.6                      pypi_0    pypi
msgpack                   1.0.3                    pypi_0    pypi
natsort                   8.0.2                    pypi_0    pypi
ncurses                   6.2                  he6710b0_1    defaults
neptune-client            0.14.1                   pypi_0    pypi
networkx                  2.5.1                    pypi_0    pypi
ninja                     1.10.2               h4bd325d_0    conda-forge
numpy                     1.19.5           py36hfc0c790_2    conda-forge
nvidia-ml-py3             7.352.0                  pypi_0    pypi
oauthlib                  3.1.1                    pypi_0    pypi
ogb                       1.3.2                    pypi_0    pypi
olefile                   0.46               pyh9f0ad1d_1    conda-forge
opencv-python             4.5.3.56                 pypi_0    pypi
openjpeg                  2.4.0                hf7af979_0    conda-forge
openssl                   1.1.1k               h7f98852_0    conda-forge
ordereddict               1.1                      pypi_0    pypi
outdated                  0.2.1                    pypi_0    pypi
packaging                 21.3                     pypi_0    pypi
panda                     0.3.1                    pypi_0    pypi
pandas                    1.1.5                    pypi_0    pypi
parso                     0.7.1                    pypi_0    pypi
pathlib2                  2.3.6                    pypi_0    pypi
pathtools                 0.1.2                    pypi_0    pypi
pexpect                   4.8.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    8.3.1            py36h2c7a002_0    defaults
pip                       21.2.2           py36h06a4308_0    defaults
pluggy                    1.0.0                    pypi_0    pypi
prettytable               2.5.0                    pypi_0    pypi
promise                   2.3                      pypi_0    pypi
prompt-toolkit            3.0.24                   pypi_0    pypi
protobuf                  3.19.1                   pypi_0    pypi
psutil                    5.8.0                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
py                        1.11.0                   pypi_0    pypi
pyaml                     21.10.1                  pypi_0    pypi
pygments                  2.11.2                   pypi_0    pypi
pyjwt                     2.3.0                    pypi_0    pypi
pynvml                    11.4.1                   pypi_0    pypi
pyparsing                 2.4.7                    pypi_0    pypi
pyrsistent                0.18.0                   pypi_0    pypi
pysocks                   1.7.1                    pypi_0    pypi
pytest                    6.2.5                    pypi_0    pypi
python                    3.6.8                h0371630_0    defaults
python-dateutil           2.8.2                    pypi_0    pypi
python_abi                3.6                     2_cp36m    conda-forge
pytorch                   1.7.1           py3.6_cuda11.0.221_cudnn8.0.5_0    pytorch
pytz                      2021.3                   pypi_0    pypi
pywavelets                1.1.1                    pypi_0    pypi
pyyaml                    5.4.1                    pypi_0    pypi
qpsolvers                 1.7.0                    pypi_0    pypi
quadprog                  0.1.10                   pypi_0    pypi
readline                  7.0                  h7b6447c_5    defaults
requests                  2.26.0                   pypi_0    pypi
requests-oauthlib         1.3.0                    pypi_0    pypi
rfc3987                   1.3.8                    pypi_0    pypi
s3transfer                0.5.0                    pypi_0    pypi
schema                    0.7.5                    pypi_0    pypi
scikit-image              0.17.2                   pypi_0    pypi
scikit-learn              0.24.2                   pypi_0    pypi
scipy                     1.5.4                    pypi_0    pypi
seaborn                   0.11.2                   pypi_0    pypi
sentry-sdk                1.4.3                    pypi_0    pypi
setuptools                52.0.0           py36h06a4308_0    defaults
shortuuid                 1.0.1                    pypi_0    pypi
simplejson                3.17.6                   pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
smmap                     5.0.0                    pypi_0    pypi
soupsieve                 2.3.1                    pypi_0    pypi
spline                    1.12                     pypi_0    pypi
sqlite                    3.33.0               h62c20be_0    defaults
strict-rfc3339            0.7                      pypi_0    pypi
subprocess32              3.5.4                    pypi_0    pypi
swagger-spec-validator    2.7.4                    pypi_0    pypi
tabulate                  0.8.9                    pypi_0    pypi
tensorboardx              2.4.1                    pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
threadpoolctl             2.2.0                    pypi_0    pypi
tifffile                  2020.9.3                 pypi_0    pypi
timm                      0.3.2                    pypi_0    pypi
tk                        8.6.10               hbc83047_0    defaults
toml                      0.10.2                   pypi_0    pypi
torchaudio                0.7.2                      py36    pytorch
torchvision               0.8.2                py36_cu110    pytorch
tqdm                      4.62.2                   pypi_0    pypi
traitlets                 4.3.3                    pypi_0    pypi
typing                    3.7.4.3                  pypi_0    pypi
typing_extensions         3.10.0.0           pyha770c72_0    conda-forge
urllib3                   1.26.7                   pypi_0    pypi
wandb                     0.12.6                   pypi_0    pypi
wcwidth                   0.2.5                    pypi_0    pypi
webcolors                 1.11.1                   pypi_0    pypi
websocket-client          1.2.3                    pypi_0    pypi
wheel                     0.37.0             pyhd3eb1b0_0    defaults
wilds                     1.2.2                    pypi_0    pypi
xz                        5.2.5                h7b6447c_0    defaults
yacs                      0.1.8                    pypi_0    pypi
yapf                      0.32.0                   pypi_0    pypi
yaspin                    2.1.0                    pypi_0    pypi
zipp                      3.6.0                    pypi_0    pypi
zlib                      1.2.11               h7b6447c_3    defaults
zstd                      1.4.9                ha95c52a_0    conda-forge
